pkg/ecsdeploy/ecs_deploy_test.go

```diff
@@ -0,0 +1,299 @@
+	assert.Nil(suite.T(), resp)
+	assert.Error(suite.T(), err)
+
+	suite.ecsMockClient.AssertExpectations(suite.T())
+}
+
+////////////////////
+// Testing RegisterUpdatedTaskDefinition
+////////////////////
+
+type RegisterUpdatedTaskDefinitionSuite struct {
+	suite.Suite
+	ecsMockClient *ecsMock
+	e             ECSClusterServiceDeployer
+	containerMap  map[string]map[string]string
+}
+
+func (suite *RegisterUpdatedTaskDefinitionSuite) SetupTest() {
+	var logger *zap.Logger // not for sure what to do about zap
+	logger, _ = zap.NewDevelopment()
+	suite.ecsMockClient = new(ecsMock)
+	suite.e = ECSClusterServiceDeployer{
+		Logger:     logger,
+		ECSCluster: "mycluster",
+		ECSService: "myservice",
+		ECSClient:  suite.ecsMockClient,
+	}
+	suite.containerMap = map[string]map[string]string{
+		"atlantis":            {"image": "updated/imagepath:latest"},
+		"nginxbutwithracoons": {"image": "nginxbutwithracoons/imagepath:latest"},
+	}
+}
+
+func TestRegisterUpdatedTaskDefinition(t *testing.T) {
+	suite.Run(t, new(RegisterUpdatedTaskDefinitionSuite))
+}
+
+// TestUpdateServiceError tests the case where the ECS API call throws an error
+func (suite *RegisterUpdatedTaskDefinitionSuite) TestRegisterUpdatedTaskDefinition() {
+	output := ecs.RegisterTaskDefinitionOutput{}
+
+	// Ensure the search works correctly
+	suite.ecsMockClient.On("RegisterTaskDefinition",
+		mock.MatchedBy(
+			func(input *ecs.RegisterTaskDefinitionInput) bool {
+				return len(input.ContainerDefinitions) == 2
+			},
+		),
+	).Return(&output, nil)
+
+	taskDef := ecs.TaskDefinition{
+		ContainerDefinitions: make([]*ecs.ContainerDefinition, 0),
+	}
+	for container := range suite.containerMap {
+		c := ecs.ContainerDefinition{}
+		c.SetName(container)
+		taskDef.ContainerDefinitions = append(taskDef.ContainerDefinitions, &c)
+	}
+
+	resp, err := suite.e.RegisterUpdatedTaskDefinition(&taskDef, suite.containerMap)
+	assert.Nil(suite.T(), resp)
+	assert.Nil(suite.T(), err)
+
+	suite.ecsMockClient.AssertExpectations(suite.T())
+}
+
+// TestRegisterUpdatedTaskDefinitionBadContainers test mismatched input
+func (suite *RegisterUpdatedTaskDefinitionSuite) TestRegisterUpdatedTaskDefinitionBadContainers() {
+	output := ecs.RegisterTaskDefinitionOutput{}
+
+	// Ensure the search works correctly
+	suite.ecsMockClient.On("RegisterTaskDefinition",
+		mock.MatchedBy(
+			func(input *ecs.RegisterTaskDefinitionInput) bool {
+				return len(input.ContainerDefinitions) == 2
+			},
+		),
+	).Return(&output, nil)
+
+	taskDef := ecs.TaskDefinition{
+		ContainerDefinitions: make([]*ecs.ContainerDefinition, 0),
+	}
+	for container := range suite.containerMap {
+		c := ecs.ContainerDefinition{}
+		c.SetName(container)
+		taskDef.ContainerDefinitions = append(taskDef.ContainerDefinitions, &c)
+	}
+
+	taskDef.ContainerDefinitions[0].SetName("blah")
+
+	resp, err := suite.e.RegisterUpdatedTaskDefinition(&taskDef, suite.containerMap)
+	assert.Nil(suite.T(), resp)
```

Comment:
Similar to my last comment, I don't think I understand the assertions in this test ....